src/Lean/Meta/WHNF.lean

```diff
@@ -75,7 +75,7 @@ def isAuxDef (constName : Name) : MetaM Bool := do
 -- ===========================
 
 private def getFirstCtor (d : Name) : MetaM (Option Name) := do
-  let some (ConstantInfo.inductInfo { ctors := ctor::_, ..}) ← getUnfoldableConstNoEx? d |
+  let some (ConstantInfo.inductInfo { ctors := ctor::_, ..}) := (← getEnv).find? d |
```

Comment:
This is a good change, and we should merge it as a 'fix' in a separate PR. Using `getUnfoldedConstNoEx?` here looks very strange, so I decided to trace its origin. It originates from Lean 3, where a function for retrieving a constant without throwing an exception was used. Since then, the code has been modified and refactored a few  times, but this line was never addressed and has become increasingly odd.